_delphi_utils_python/delphi_utils/logger.py

```diff
@@ -1,21 +1,44 @@
-"""Structured logger utility for creating JSON logs in Delphi pipelines."""
+"""Structured logger utility for creating JSON logs."""
+
+# the Delphi group uses two ~identical versions of this file.
+# try to keep them in sync with edits, for sanity.
+#   https://github.com/cmu-delphi/covidcast-indicators/blob/main/_delphi_utils_python/delphi_utils/logger.py
+#   https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/common/logger.py
+
+import contextlib
 import logging
+import multiprocessing
+import os
+import structlog
 import sys
 import threading
-import structlog
+from traceback import format_exception
 
 
 def handle_exceptions(logger):
     """Handle exceptions using the provided logger."""
-    def exception_handler(etype, value, traceback):
+
+    def exception_handler(scope, etype, value, traceback):
         logger.exception("Top-level exception occurred",
-                         exc_info=(etype, value, traceback))
+                         scope=scope, exc_info=(etype, value, traceback))
+
+    def sys_exception_handler(etype, value, traceback):
+        exception_handler("sys", etype, value, traceback)
 
-    def multithread_exception_handler(args):
-        exception_handler(args.exc_type, args.exc_value, args.exc_traceback)
+    def threading_exception_handler(args):
+        if args.exc_type == SystemExit and args.exc_value.code == 0:
+            # `sys.exit(0)` is considered "successful termination":
+            #   https://docs.python.org/3/library/sys.html#sys.exit
+            logger.debug("normal thread exit", thread=args.thread,
+                        stack="".join(
+                            format_exception(
+                                args.exc_type, args.exc_value, args.exc_traceback)))
```

Comment:
Lines like this seem like a good reason to loosen the line length requirement for pylint like you mentioned on Slack :)